torch/csrc/api/include/torch/nn/module.h

```diff
@@ -511,6 +514,26 @@ class TORCH_API Module : public std::enable_shared_from_this<Module> {
   /// with `name` an exception is thrown.
   void unregister_module(const std::string& name);
 
+  /// Registers a forward pre-hook on the module.
+  ///
+  /// The hook will be called every time before `forward` is invoked.
+  /// It should have the following signature:
+  ///
+  ///     hook(module, input) -> void
+  ///
+  /// The hook can modify the input.
```

Comment:
Why did you made this choice given that the Python version had the opposite behavior?